src/shogun/lib/SGVector.h

```diff
@@ -522,13 +516,20 @@ template<class T> class SGVector : public SGReferencedData
 		/** needs to be overridden to free data */
 		virtual void free_data();
 
+	private:
+		void on_gpu_check() const
+		{
+			if (on_gpu())
+				SG_SERROR("You called a direct memory access method when data is in GPU memory.\n");
```

Comment:
Phrase indirect
"Direct memory access not possible when data is in GPU memory.\n"